csrc/multidevice/communicator.h

```diff
@@ -126,6 +139,10 @@ class Communicator {
   }
 
  private:
+  Communicator(
+      CommunicatorBackend backend = comm_backend_default,
+      RankType server_local_rank = comm_server_local_rank_default);
```

Comment:
nitpick: I feel at this point, we are not using the arguments for the constructor any more.

Reply:
Indeed we always use default values currently. I am not opposed to removing those arguments if you think that's preferable. Anyway, IMO, the Communicator's interface's final state will very much depend on what will be the multi-device API with frontend -- for now, I think it is too early to have a precise idea about that

Reply:
> nitpick: I feel at this point, we are not using the arguments for the constructor any more.

Good point. I'll give that a shot in a separate PR. 

Singleton implies that there can only be one instance, and therefore any configs are essentially global. So likely getInstance or the constructor would have to parse flags and decide the config. getInstance is a slightly better place to do so because if later we decide to allow multiple Communicator instances we don't have to change the constructor.